sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -933,23 +933,6 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
     val properties = Option(ctx.tablePropertyList).map(visitPropertyKeyValues).getOrElse(Map.empty)
     val selectQuery = Option(ctx.query).map(plan)
 
-    // Ensuring whether no duplicate name is used in table definition
-    val colNames = dataCols.map(_.name)
-    if (colNames.length != colNames.distinct.length) {
-      val duplicateColumns = colNames.groupBy(identity).collect {
-        case (x, ys) if ys.length > 1 => "\"" + x + "\""
-      }
-      operationNotAllowed(s"Duplicated column names found in table definition of $name: " +
-        duplicateColumns.mkString("[", ",", "]"), ctx)
-    }
-
-    // For Hive tables, partition columns must not be part of the schema
```

Comment:
wait. I found a conflict here. 

For Hive Tables, the following query is not allowed. 
```
CREATE TABLE tab1 (key INT, value STRING) PARTITIONED BY (key INT)
```
However, for data source tables, this query is allowed with a minor change. 
```
CREATE TABLE tab1 (key INT, value STRING) USING json PARTITIONED BY (key)
```

Thus, they are different regarding whether partitioning columns should or should not be included in data columns.